File: scala/play2/akka_scheduler_sample/app/controllers/Application.scala
package controllers

import play.api._
import play.api.Play.current
import play.api.mvc._
import play.api.libs.concurrent._

import akka.util.duration._

object Application extends Controller {

  def index = Action {
    Ok(views.html.index())
  }

  def addJob = Action { req =>
    // 1秒後に非同期処理を実行
    Akka.system.scheduler.scheduleOnce(0 seconds) {
        println("start sleep : " + req)
        Thread.sleep(5000)
        println("stop sleep : " + req)
    }

    Ok("added job")
  }

}

File: java/play2/async_sample/app/controllers/Application.java
package controllers;

import play.*;
import play.mvc.*;
import play.libs.Akka;

import views.html.*;

import akka.util.FiniteDuration;

public class Application extends Controller {

	public static Result index() {
		return ok(index.render());
	}

	public static Result jobAdd() {
		final Http.Request req = request();

		Akka.system().scheduler().scheduleOnce(
			new FiniteDuration(0, "seconds"),
			new Runnable() {
				public void run() {
					System.out.println("start sleep : " + req);

					try {
						Thread.currentThread().sleep(5000);
					} catch (Exception ex) {
						ex.printStackTrace();
					}

					System.out.println("stop sleep : " + req);
				}
			}
		);

		return ok("job added");
	}

}